### ISSUE TITLE
Could not install WWDR certificate

### ISSUE THREAD

**Issue by dev-loic** [fastlane/fastlane#20960]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Hello everyone 👋🏻

We are using circle ci with fastlane 2.211.0. 
Randomly when we do the matching certificates, the CI fails with the following error `Could not install WWDR certificate`.
I say randomly because it does not happen each time and the logs are not relevant both ways. 

Thanks in advance for your insights

##### Command executed

`match`

##### Complete output when running fastlane, including the stack trace and command used

Here is the output **when the CI fails**
<details>
  <pre> 
chruby: unknown Ruby: ruby-2.6.8
Your RubyGems version (3.0.3.1) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [✔] 🚀 
+------------------------------------------+---------+--------------------------+
|                                 Used plugins                                  |
+------------------------------------------+---------+--------------------------+
| Plugin                                   | Version | Action                   |
+------------------------------------------+---------+--------------------------+
| fastlane-plugin-diawi                    | 2.1.0   | diawi                    |
| fastlane-plugin-clean_testflight_testers | 0.3.0   | clean_testflight_testers |
+------------------------------------------+---------+--------------------------+

[15:43:52]: Sending anonymous analytics information
[15:43:52]: Learn more at https://docs.fastlane.tools/#metrics
[15:43:52]: No personal or sensitive data is sent.
[15:43:52]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[15:43:53]: ----------------------------------------
[15:43:53]: --- Step: Verifying fastlane version ---
[15:43:53]: ----------------------------------------
[15:43:53]: Your fastlane version 2.211.0 matches the minimum requirement of 2.114.0  ✅
[15:43:53]: ------------------------------
[15:43:53]: --- Step: default_platform ---
[15:43:53]: ------------------------------
[15:43:53]: Driving the lane 'ios build_and_release_ci' 🚀
[15:43:53]: -----------------------------
[15:43:53]: --- Step: setup_circle_ci ---
[15:43:53]: -----------------------------
[15:43:53]: Skipping Log Path setup as FL_OUTPUT_DIR is unset
[15:43:53]: Creating temporary keychain: "fastlane_tmp_keychain".
[15:43:53]: $ security list-keychains -d user
[15:43:53]: ▸ "/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db"
[15:43:53]: Found keychain '/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[15:43:53]: Enabling match readonly mode.
[15:43:53]: ----------------------------------
[15:43:53]: --- Step: get_info_plist_value ---
[15:43:53]: ----------------------------------
[15:43:53]: ----------------------------------
[15:43:53]: --- Step: get_info_plist_value ---
[15:43:53]: ----------------------------------
[15:43:53]: ⚙️ Start building 6.0.4 Build 1
[15:43:53]: ---------------------------------------------------
[15:43:53]: --- Step: Switch to ios post_slack_message lane ---
[15:43:53]: ---------------------------------------------------
[15:43:53]: Cruising over to lane 'ios post_slack_message' 🚖
[15:43:53]: -------------------
[15:43:53]: --- Step: slack ---
[15:43:53]: -------------------
[15:43:53]: Successfully sent Slack notification
[15:43:53]: Cruising back to lane 'ios build_and_release_ci' 🚘
[15:43:53]: ---------------------------------------------------------------
[15:43:53]: --- Step: Switch to ios test_app_store_connect_api_key lane ---
[15:43:53]: ---------------------------------------------------------------
[15:43:53]: Cruising over to lane 'ios test_app_store_connect_api_key' 🚖
[15:43:53]: ---------------------------------------
[15:43:53]: --- Step: app_store_connect_api_key ---
[15:43:53]: ---------------------------------------
[15:43:53]: Cruising back to lane 'ios build_and_release_ci' 🚘
[15:43:53]: 🤝 Match for com.toto
[15:43:53]: -------------------
[15:43:53]: --- Step: match ---
[15:43:53]: -------------------
[15:43:53]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+

+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto"]                             |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[15:43:53]: Cloning remote git repo...
[15:43:53]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[15:43:54]: Checking out branch master...
[15:43:54]: 🔓  Successfully decrypted certificates repo
[15:43:54]: Installing certificate...
[15:43:54]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[15:43:55]: --------------------------------------------
[15:43:55]: --- Step: mv Appfile.tmp Appfile || true ---
[15:43:55]: --------------------------------------------
[15:43:55]: $ mv Appfile.tmp Appfile || true
[15:43:55]: ▸ mv: rename Appfile.tmp to Appfile: No such file or directory
[15:43:55]: ---------------------------------------------------------
[15:43:55]: --- Step: Switch to ios post_slack_error_message lane ---
[15:43:55]: ---------------------------------------------------------
[15:43:55]: Cruising over to lane 'ios post_slack_error_message' 🚖
[15:43:55]: -------------------
[15:43:55]: --- Step: slack ---
[15:43:55]: -------------------
[15:43:55]: Successfully sent Slack notification
[15:43:55]: Cruising back to lane 'ios build_and_release_ci' 🚘
+-----------------------------------+--------------------------------------------------------+
|                                        Lane Context                                        |
+-----------------------------------+--------------------------------------------------------+
| DEFAULT_PLATFORM                  | ios                                                    |
| PLATFORM_NAME                     | ios                                                    |
| LANE_NAME                         | ios build_and_release_ci                               |
| KEYCHAIN_PATH                     | ~/Library/Keychains/fastlane_tmp_keychain              |
| ORIGINAL_DEFAULT_KEYCHAIN         | "/Users/distiller/Library/Keychains/login.keychain-db" |
| GET_INFO_PLIST_VALUE_CUSTOM_VALUE | 1                                                      |
+-----------------------------------+--------------------------------------------------------+
[15:43:55]: Could not install WWDR certificate

+------+---------------------------------------------------+-------------+
|                            fastlane summary                            |
+------+---------------------------------------------------+-------------+
| Step | Action                                            | Time (in s) |
+------+---------------------------------------------------+-------------+
| 1    | Verifying fastlane version                        | 0           |
| 2    | default_platform                                  | 0           |
| 3    | setup_circle_ci                                   | 0           |
| 4    | get_info_plist_value                              | 0           |
| 5    | get_info_plist_value                              | 0           |
| 6    | Switch to ios post_slack_message lane             | 0           |
| 7    | slack                                             | 0           |
| 8    | Switch to ios test_app_store_connect_api_key lane | 0           |
| 9    | app_store_connect_api_key                         | 0           |
| 💥   | match                                             | 1           |
| 11   | mv Appfile.tmp Appfile || true                    | 0           |
| 12   | Switch to ios post_slack_error_message lane       | 0           |
| 13   | slack                                             | 0           |
+------+---------------------------------------------------+-------------+

[15:43:55]: fastlane finished with errors

[!] Could not install WWDR certificate

Exited with code exit status 1
 </pre>
</details>

Here is the output when **not having the issue**

<details>
<pre>
chruby: unknown Ruby: ruby-2.6.8
Your RubyGems version (3.0.3.1) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [✔] 🚀 
+------------------------------------------+---------+--------------------------+
|                                 Used plugins                                  |
+------------------------------------------+---------+--------------------------+
| Plugin                                   | Version | Action                   |
+------------------------------------------+---------+--------------------------+
| fastlane-plugin-diawi                    | 2.1.0   | diawi                    |
| fastlane-plugin-clean_testflight_testers | 0.3.0   | clean_testflight_testers |
+------------------------------------------+---------+--------------------------+

[10:04:59]: Sending anonymous analytics information
[10:04:59]: Learn more at https://docs.fastlane.tools/#metrics
[10:04:59]: No personal or sensitive data is sent.
[10:04:59]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[10:05:00]: ----------------------------------------
[10:05:00]: --- Step: Verifying fastlane version ---
[10:05:00]: ----------------------------------------
[10:05:00]: Your fastlane version 2.211.0 matches the minimum requirement of 2.114.0  ✅
[10:05:00]: ------------------------------
[10:05:00]: --- Step: default_platform ---
[10:05:00]: ------------------------------
[10:05:00]: Driving the lane 'ios build_and_release_ci' 🚀
[10:05:00]: -----------------------------
[10:05:00]: --- Step: setup_circle_ci ---
[10:05:00]: -----------------------------
[10:05:00]: Skipping Log Path setup as FL_OUTPUT_DIR is unset
[10:05:00]: Creating temporary keychain: "fastlane_tmp_keychain".
[10:05:00]: $ security list-keychains -d user
[10:05:00]: ▸ "/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db"
[10:05:00]: Found keychain '/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[10:05:00]: Enabling match readonly mode.
[10:05:00]: ----------------------------------
[10:05:00]: --- Step: get_info_plist_value ---
[10:05:00]: ----------------------------------
[10:05:00]: ----------------------------------
[10:05:00]: --- Step: get_info_plist_value ---
[10:05:00]: ----------------------------------
[10:05:00]: ⚙️ Start building 6.0.5 Build 2
[10:05:00]: ---------------------------------------------------
[10:05:00]: --- Step: Switch to ios post_slack_message lane ---
[10:05:00]: ---------------------------------------------------
[10:05:00]: Cruising over to lane 'ios post_slack_message' 🚖
[10:05:00]: -------------------
[10:05:00]: --- Step: slack ---
[10:05:00]: -------------------
[10:05:01]: Successfully sent Slack notification
[10:05:01]: Cruising back to lane 'ios build_and_release_ci' 🚘
[10:05:01]: ---------------------------------------------------------------
[10:05:01]: --- Step: Switch to ios test_app_store_connect_api_key lane ---
[10:05:01]: ---------------------------------------------------------------
[10:05:01]: Cruising over to lane 'ios test_app_store_connect_api_key' 🚖
[10:05:01]: ---------------------------------------
[10:05:01]: --- Step: app_store_connect_api_key ---
[10:05:01]: ---------------------------------------
[10:05:01]: Cruising back to lane 'ios build_and_release_ci' 🚘
[10:05:01]: 🤝 Match for com.toto
[10:05:01]: -------------------
[10:05:01]: --- Step: match ---
[10:05:01]: -------------------
[10:05:01]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeamTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+

+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto"]                             |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[10:05:01]: Cloning remote git repo...
[10:05:01]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[10:05:02]: Checking out branch master...
[10:05:02]: 🔓  Successfully decrypted certificates repo
[10:05:02]: Installing certificate...
[10:05:02]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[10:05:03]: There are no local code signing identities found.
You can run `security find-identity -v -p codesigning fastlane_tmp_keychain` to get this output.
This Stack Overflow thread has more information: https://stackoverflow.com/q/35390072/774.
(Check in Keychain Access for an expired WWDR certificate: https://stackoverflow.com/a/35409835/774 has more info.)
[10:05:03]: Setting key partition list... (this can take a minute if there are a lot of keys installed)
[10:05:03]: security: SecItemCopyMatching: The specified item could not be found in the keychain.
[10:05:03]: Setting key partition list... (this can take a minute if there are a lot of keys installed)

+-------------------+--------------------------------------+
|                  Installed Certificate                   |
+-------------------+--------------------------------------+
| User ID           | ABCDEFGHI7                           |
| Common Name       | Apple Distribution: TOTO (ABCDEFGHI7) |
| Organisation Unit | ABCDEFGHI7                           |
| Organisation      | TOTO                                  |
| Country           | US                                   |
| Start Datetime    | 2022-03-02 08:32:09 UTC              |
| End Datetime      | 2023-03-02 08:32:08 UTC              |
+-------------------+--------------------------------------+

[10:05:03]: Installing provisioning profile...

+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
|                                                                              Installed Provisioning Profile                                                                              |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
| Parameter           | Environment Variable                            | Value                                                                                                            |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
| App Identifier      |                                                 | com.toto                                                                                               |
| Type                |                                                 | appstore                                                                                                         |
| Platform            |                                                 | ios                                                                                                              |
| Profile UUID        | sigh_com.toto_appstore                  | 6ad3dd91-290f-43b5-a06c-8a6e5ba9586f                                                                           |
| Profile Name        | sigh_com.toto_appstore_profile-name     | match AppStore com.toto                                                                                  |
| Profile Path        | sigh_com.toto_appstore_profile-path     | /Users/distiller/Library/MobileDevice/Provisioning Profiles/6ad3dd91-290f-43b5-a06c-8a6e5ba9586f.mobileprovision |
| Development Team ID | sigh_com.toto_appstore_team-id          | ABCDEFGHI7                                                                                                       |
| Certificate Name    | sigh_com.toto_appstore_certificate-name | Apple Distribution: TOTO (ABCDEFGHI7)                                                                             |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+

[10:05:03]: All required keys, certificates and provisioning profiles are installed 🙌
[10:05:03]: Setting Provisioning Profile type to 'app-store'
[10:05:03]: 🤝 Match for com.toto.Notification
[10:05:03]: -------------------
[10:05:03]: --- Step: match ---
[10:05:03]: -------------------
[10:05:03]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+

+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto.notification"]           |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[10:05:03]: Cloning remote git repo...
[10:05:03]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[10:05:04]: Checking out branch master...
[10:05:04]: 🔓  Successfully decrypted certificates repo
[10:05:04]: Installing certificate...
[10:05:04]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[10:05:04]: ▸ -----BEGIN CERTIFICATE-----
.......
</pre>
</details>

### Environment
We are using fastlane `2.211.0`

**Comment by dev-loic** [comment]
Hey Fastlane team, 
I use the `verbose: true` arg on `match` command and it seems it fails on the G6 certificate import 
<details>
<pre>DEBUG [2023-01-03 17:28:18.95]: Installing WWDR Cert: curl -f -o /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/AppleWWDRCAG6.cer20230103-4270-11yw6p8 https://www.apple.com/certificateauthority/AppleWWDRCAG6.cer && security import /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/AppleWWDRCAG6.cer20230103-4270-11yw6p8 -k /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
INFO [2023-01-03 17:28:19.00]: ▸ 100   794  100   794    0     0  30909      0 --:--:-- --:--:-- --:--:-- 39700
INFO [2023-01-03 17:28:19.00]: ▸ security: SecKeychainItemImport: Unknown format in import.
DEBUG [2023-01-03 17:28:19.00]: Failed to install WWDR Certificate, checking output to see why </pre>
</details>
While with the exact same circle ci workflow, on the same develop branch trigger I've got 
<details>
<pre>DEBUG [2023-01-03 17:26:49.76]: Installing WWDR Cert: curl -f -o /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/AppleWWDRCAG6.cer20230103-4271-3gxe8q https://www.apple.com/certificateauthority/AppleWWDRCAG6.cer && security import /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/AppleWWDRCAG6.cer20230103-4271-3gxe8q -k /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
INFO [2023-01-03 17:26:49.81]: ▸ 1 certificate imported.
INFO [2023-01-03 17:26:49.81]: ▸ 100   794  100   794    0     0  30197      0 --:--:-- --:--:-- --:--:-- 37809</pre>
</details>

And it looks like it the result of this line https://github.com/fastlane/fastlane/blob/6ff1c49bd90bdc29a94b6f99f568cf86fffc376d/fastlane_core/lib/fastlane_core/cert_checker.rb#L160

**Comment by Arjan-Zuidema** [comment]
Same issue here, running on Github Actions

**Comment by yksingh12** [comment]
facing the same issue on GitHub Actions

**Comment by Arjan-Zuidema** [comment]
Our issue has been resolved by itself, seems like it is an Apple problem?

**Comment by yksingh12** [comment]
Some intermittent issue, resolved itself on next build

**Comment by piellarda** [comment]
Happens to us as well every few builds (CircleCI).

**Comment by juan-vg** [comment]
Same for me. Fortunately I have debug enabled, and the most common error message I get is the following one:
```
curl: (92) HTTP/2 stream 1 was not closed cleanly before end of the underlying stream
```

However, sometimes I get the following one:
```
Installing WWDR Cert: curl -f -o /var/folders/gh/5h9l2j0n4klg38n4bhn256yc0333gm/T/AppleWWDRCAG3.cer20230109-941-zd7vp7 https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer && security import /var/folders/gh/5h9l2j0n4klg38n4bhn256yc0333gm/T/AppleWWDRCAG3.cer20230109-941-zd7vp7 -k /Users/myuser/Library/Keychains/login.keychain-db

INFO [2023-01-09 18:28:34.08]: ▸ 100  1109  100  1109    0     0   7648      0 --:--:-- --:--:-- --:--:--  7809
INFO [2023-01-09 18:28:34.08]: ▸ security: SecKeychainItemImport: Unknown format in import.
```

**Comment by juan-vg** [comment]
After several tests I can conclude downgrading to `2.210.1` solves the issue. Additionally I have discovered that both versions work fine when only one Fastlane lane is run. When two or more lanes are run in parallel is when `2.211.0` fails (at least in my case).

I've then checked all commits between both versions and the only one related to this topic is 4b3008914ff809850d7b8bf772e1a3363213ed18. The issue must be there.

**Comment by dev-loic** [comment]
We also downgrade to `2.210.1` and until now this issue did not pop up anymore

**Comment by akselilukkarila** [comment]
This happens to us quite often in CI and is really annoying 😞

**Comment by dcacenabes** [comment]
Same is happening for us.

**Comment by aprilmintacpineda** [comment]
Currently happening to me on Github Actions. Works fine if I run it locally.

<details>

<summary>full logs</summary>

```
[06:56:26]: Sending anonymous analytics information
[06:56:26]: Learn more at https://docs.fastlane.tools/#metrics
[06:56:26]: No personal or sensitive data is sent.
[06:56:26]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[06:56:26]: ------------------------------
[06:56:26]: --- Step: default_platform ---
[06:56:26]: ------------------------------
[06:56:26]: Driving the lane 'ios productionRelease' 🚀
[06:56:26]: ----------------------
[06:56:26]: --- Step: setup_ci ---
[06:56:26]: ----------------------
[06:56:26]: Creating temporary keychain: "fastlane_tmp_keychain".
[06:56:26]: $ security list-keychains -d user
[06:56:26]: ▸ "/Users/runner/Library/Keychains/fastlane_tmp_keychain-db"
[06:56:26]: Found keychain '/Users/runner/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[06:56:26]: Enabling match readonly mode.
[06:56:26]: ---------------------------------------
[06:56:26]: --- Step: app_store_connect_api_key ---
[06:56:26]: ---------------------------------------
[06:56:27]: -------------------
[06:56:27]: --- Step: match ---
[06:56:27]: -------------------
[06:56:27]: Successfully loaded '/Users/runner/work/myapp-react-native/myapp-react-native/ios/fastlane/Matchfile' 📄

+----------------+----------------------------------------------------------------------------------------+
|                               Detected Values from './fastlane/Matchfile'                               |
+----------------+----------------------------------------------------------------------------------------+
| git_url        | git@github.com:aprilmintacpineda/myapp-fastlane-match.git                              |
| storage_mode   | git                                                                                    |
| type           | appstore                                                                               |
| app_identifier | ["com.aprmp.myapp", "com.aprmp.myapp.stg", "com.aprmp.myapp.dev"]                      |
| username       | ***                                                                                    |
+----------------+----------------------------------------------------------------------------------------+


+----------------------------------------+----------------------------------------------------------------------------------------+
|                                                    Summary for match 2.211.0                                                    |
+----------------------------------------+----------------------------------------------------------------------------------------+
| api_key                                | ********                                                                               |
| readonly                               | true                                                                                   |
| type                                   | appstore                                                                               |
| generate_apple_certs                   | true                                                                                   |
| skip_provisioning_profiles             | false                                                                                  |
| app_identifier                         | ["com.aprmp.myapp", "com.aprmp.myapp.stg", "com.aprmp.myapp.dev"]                      |
| username                               | ***                                                                                    |
| team_id                                | ***                                                                                    |
| storage_mode                           | git                                                                                    |
| git_url                                | git@github.com:aprilmintacpineda/myapp-fastlane-match.git                              |
| git_branch                             | master                                                                                 |
| shallow_clone                          | false                                                                                  |
| clone_branch_directly                  | false                                                                                  |
| skip_google_cloud_account_confirmation | false                                                                                  |
| keychain_name                          | fastlane_tmp_keychain                                                                  |
| force                                  | false                                                                                  |
| force_for_new_devices                  | false                                                                                  |
| include_mac_in_profiles                | false                                                                                  |
| include_all_certificates               | false                                                                                  |
| force_for_new_certificates             | false                                                                                  |
| skip_confirmation                      | false                                                                                  |
| safe_remove_certs                      | false                                                                                  |
| skip_docs                              | false                                                                                  |
| platform                               | ios                                                                                    |
| derive_catalyst_app_identifier         | false                                                                                  |
| fail_on_name_taken                     | false                                                                                  |
| skip_certificate_matching              | false                                                                                  |
| skip_set_partition_list                | false                                                                                  |
| verbose                                | false                                                                                  |
+----------------------------------------+----------------------------------------------------------------------------------------+

[06:56:27]: Cloning remote git repo...
[06:56:27]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[06:56:29]: Checking out branch master...
[06:56:29]: 🔓  Successfully decrypted certificates repo
[06:56:29]: Installing certificate...
[06:56:29]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/runner/Library/Keychains/fastlane_tmp_keychain-db
+---------------------------+-----------------------------------------------------+
|                                  Lane Context                                   |
+---------------------------+-----------------------------------------------------+
| DEFAULT_PLATFORM          | ios                                                 |
| PLATFORM_NAME             | ios                                                 |
| LANE_NAME                 | ios productionRelease                               |
| KEYCHAIN_PATH             | ~/Library/Keychains/fastlane_tmp_keychain           |
| ORIGINAL_DEFAULT_KEYCHAIN | "/Users/runner/Library/Keychains/login.keychain-db" |
+---------------------------+-----------------------------------------------------+
[06:56:30]: Could not install WWDR certificate


[!] Could not install WWDR certificate
+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | default_platform          | 0           |
| 2    | setup_ci                  | 0           |
| 3    | app_store_connect_api_key | 0           |
| 💥   | match                     | 3           |
+------+---------------------------+-------------+

[06:56:30]: fastlane finished with errors
Error: Process completed with exit code 1.
```

</details>

**Comment by aprilmintacpineda** [comment]
> I'm struggling to reproduce it on my test repo, but I have seen this occur sporadically on Bitrise.
> 
> Some of this is speculation, but from what I can see:
> 
> * Before 2.211.0 the success check in `cert_checker` was looking at the wrong process (this is why it was changed according to the commit [4b30089#diff-605d869e313da5ef9f0c402845a4911eed5f8afc4ef45382210f237fceae498aL169](https://github.com/fastlane/fastlane/commit/4b3008914ff809850d7b8bf772e1a3363213ed18#diff-605d869e313da5ef9f0c402845a4911eed5f8afc4ef45382210f237fceae498aL169)), which then meant all failures in `cert_checker` were being ignored
> * Some machines have some WWDR certificates pre-installed, so even though `cert_checker` was failing to install them, other parts of Fastlane kept on going because they didn't rely upon the missing certs
> * The HTTP/2 error is a transient error from the server, which is why not everyone is seeing it all of the time
> 
> Using Curl through a shell call feels dirty, but I don't think replacing it with Faraday (or something else) will necessarily solve this.
> 
> Backing out the 2.211.0 change to the success check is papering over another issue, and only works by pure luck.
> 
> In the Curl command we could add `--http1.1` and/or `--retry 3 --retry-all-errors` to try and alleviate it in the short term.
> 
> Another solution would be to add an option to Match to skip the WWDR cert installation, but I think this will take a lot of unpicking, and possibly just creates further problems in the future when the WWDR cert is renewed again.
> 
> Without being able to reproduce it I'm hesistant to just chuck a PR in and have all the Match users alpha-test it.

Actually I also can't reproduce the error on my machine, it happens to Github Actions consistently though, maybe you can try there?

**Comment by simplenotezy** [comment]
> * I can confirm that downgrading to version `2.210.1` seems to get rid of the problem.

Same here. FYI for anybody who'd like to know how, simply edit `Gemfile`:

```diff
source "https://rubygems.org"

-# gem "fastlane"
+gem "fastlane", "= 2.210.1"
gem "fastlane-plugin-flutter_version", git: "https://github.com/tianhaoz95/fastlane-plugin-flutter-version"
plugins_path = File.join(File.dirname(__FILE__), '.', 'Pluginfile')
eval_gemfile(plugins_path) if File.exist?(plugins_path)

**Comment by juan-vg** [comment]
@chedabob maybe the installed Curl version is a factor here, or even the way Fastlane was installed (Ruby [Bundler] vs. Brew). If the previous way of checking Curl results was not properly checking them, perhaps/probably Curl was already failing before without noticing. If so, stopping relying on a system binary which is not under Fastlane's control (which could also be at different versions for each user) by replacing it with Faraday (or something else) could actually solve it. At least it will provide a homogeneous scenario for everybody. Additionally, removing the dependency on a system package which ―as I said― is not under Fastlane's control looks like something that should be addressed ASAP. 

Either way, right now adding those flags will probably mitigate this issue as a quick workaround. On top it will help to test if the HTTP/2 protocol was the real problem or if it's somewhere else.

**Comment by chedabob** [comment]
If anyone would like to test, I've put a branch up that migrates from Curl to Faraday

Replace the `fastlane` line in your Gemfile with this

`gem 'fastlane', git: 'https://github.com/chedabob/fastlane', branch: 'CertCheckerFaraday'`

One of my colleagues was also seeing this error so I will get them to verify this week.

**Comment by ptmt** [comment]
I tried but still get `Could not install WWDR certificate` 

```
[10:50:39 ] Using fastlane 2.211.0 from https://github.com/chedabob/fastlane (at CertCheckerFaraday@061da97)
[10:50:43]: Installing certificate...
  [10:50:43]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Library/Keychains/System.keychain

[10:50:43 ]
  [10:50:43]: ▸ -----BEGIN CERTIFICATE-----
<certificate here>
  [10:50:43]: ▸ -----END CERTIFICATE-----
  ```

**Comment by chedabob** [comment]
> I tried but still get `Could not install WWDR certificate`
> 
> ```
> [10:50:39 ] Using fastlane 2.211.0 from https://github.com/chedabob/fastlane (at CertCheckerFaraday@061da97)
> [10:50:43]: Installing certificate...
> [10:50:43]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Library/Keychains/System.keychain
> 
> [10:50:43 ]
> [10:50:43]: ▸ -----BEGIN CERTIFICATE-----
> <certificate here>
> [10:50:43]: ▸ -----END CERTIFICATE-----
> [!] Could not install WWDR certificate
> ```

Can you run with `--verbose` and paste the whole log?

**Comment by joshdholtz** [comment]
Will try and get a new version of _fastlane_ out with a fix tonight if I can!

**Comment by ptmt** [comment]
@chedabob thanks for the tip, I see now the reason:

```
Installing WWDR Cert: security import /Users/admin/buildAgent/temp/buildTmp/AppleWWDRCAG2.cer20230209-4500-q5k519 -k /Library/Keychains/System.keychain

NFO [2023-02-09 11:26:07.81]: ▸ security: SecKeychainItemImport: Write permissions error.
  ```

**Comment by mattmapadmi** [comment]
@ptmt 

I think this might be an issue unrelated to the OP. It looks like you've only got the System keychain, which I suspect isn't writable by the fastlane process without first elevating its privlege (like a user would by entering their password/fingerprint in Keychain Access).

**Comment by ducbm051291** [comment]
I also got this issue (GitHub Action and CodeMagic), and I have a temporary solution is to run match 2 times 😅

**Comment by joshdholtz** [comment]
> I'm struggling to reproduce it on my test repo, but I have seen this occur sporadically on Bitrise.
> 
> Some of this is speculation, but from what I can see:
> 
> * Before 2.211.0 the success check in `cert_checker` was looking at the wrong process (this is why it was changed according to the commit [4b30089#diff-605d869e313da5ef9f0c402845a4911eed5f8afc4ef45382210f237fceae498aL169](https://github.com/fastlane/fastlane/commit/4b3008914ff809850d7b8bf772e1a3363213ed18#diff-605d869e313da5ef9f0c402845a4911eed5f8afc4ef45382210f237fceae498aL169)), which then meant all failures in `cert_checker` were being ignored
> * Some machines have some WWDR certificates pre-installed, so even though `cert_checker` was failing to install them, other parts of Fastlane kept on going because they didn't rely upon the missing certs
> * The HTTP/2 error is a transient error from the server, which is why not everyone is seeing it all of the time
> 
> Using Curl through a shell call feels dirty, but I don't think replacing it with Faraday (or something else) will necessarily solve this.
> 
> Backing out the 2.211.0 change to the success check is papering over another issue, and only works by pure luck.
> 
> In the Curl command we could add `--http1.1` and/or `--retry 3 --retry-all-errors` to try and alleviate it in the short term.
> 
> Another solution would be to add an option to Match to skip the WWDR cert installation, but I think this will take a lot of unpicking, and possibly just creates further problems in the future when the WWDR cert is renewed again.
> 
> Without being able to reproduce it I'm hesistant to just chuck a PR in and have all the Match users alpha-test it.

Why this is only failing occasionally on CI is way beyond me right now but... I like that there are some things we can add to the `curl` command `--http1.1` and/or `--retry 3 --retry-all-errors` that might fix this 🤷‍♂️ 

I'm going to put these behind an environment variable since I'm not able to replicate this issue but it will allow anybody facing this issue to set an environment variable like `FASTLANE_WWDR_USE_HTTP1_AND_RETRIES=true` to see if this does solve it 🤔 

I'm going to create a PR for this and then get a new release out this weekend. Would appreciate any feedback when released if this works or not... but hopefully it does 🤞

**Comment by DamienBitrise** [comment]
I think the root cause of this issue is that fastlane is looking for the certificate in the Login Keychain and the cert is installed in the System Keychain

Failing
`security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/vagrant/Library/Keychains/login.keychain-db`

Working
`security find-certificate -a -c 'Apple Worldwide Developer Relations'`

When fastlane fails to find the cert it downloads it from Apple which intermittently fails

**Comment by dcacenabes** [comment]
@joshdholtz Unfortunately it did not fix the issue. I just got the same error in our 4th CI build after merging the bump to 2.212.0 and setting the env var :(

**Comment by juan-vg** [comment]
@joshdholtz have you tried the Faraday workaround by @chedabob 

> If anyone would like to test, I've put a branch up that migrates from Curl to Faraday
> 
> Replace the `fastlane` line in your Gemfile with this
> 
> `gem 'fastlane', git: 'https://github.com/chedabob/fastlane', branch: 'CertCheckerFaraday'`
> 
> One of my colleagues was also seeing this error so I will get them to verify this week.

If not, could you please try? I've not been able to reproduce the error lately so I was unable to check it.

**Comment by combinatorial** [comment]
@joshdholtz similar with us, I updated to 2.212.0 and set the env variable and hit the issue on a run today.

**Comment by chedabob** [comment]
@combinatorial  @dcacenabes @juan-vg Are you building on Apple Silicon? Could it be you're seeing the missing keychain issue that @ptmt mentioned?

**Comment by juan-vg** [comment]
@chedabob Yep, we I'm building on Apple M1. However I can't remember the keychain was an issue. The issue happened randomly and the keychains were also in the same state for each execution (using Apple VMs freshly deployed from the same golden image each time). What I mean is that if it was a keychain issue, it should have happened every time instead of just some random times.

As you could check on mi first comment (https://github.com/fastlane/fastlane/issues/20960#issuecomment-1377403411) the issue looks different.

> Same for me: random errors when downloading WWDR certs, but I don't believe it's an Apple issue. I have another CI server with another version of fastlane and the errors there are way less frequent (or almost nonexistent).
> 
> Fortunately I have debug enabled.
> 
> The most common error message I get is the following one:
> 
> ```
> curl: (92) HTTP/2 stream 1 was not closed cleanly before end of the underlying stream
> ```
> 
> However, sometimes I get the following one:
> 
> ```
> Installing WWDR Cert: curl -f -o /var/folders/gh/5h9l2j0n4klg38n4bhn256yc0333gm/T/AppleWWDRCAG3.cer20230109-941-zd7vp7 https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer && security import /var/folders/gh/5h9l2j0n4klg38n4bhn256yc0333gm/T/AppleWWDRCAG3.cer20230109-941-zd7vp7 -k /Users/myuser/Library/Keychains/login.keychain-db
> 
> INFO [2023-01-09 18:28:34.08]: ▸ 100  1109  100  1109    0     0   7648      0 --:--:-- --:--:-- --:--:--  7809
> INFO [2023-01-09 18:28:34.08]: ▸ security: SecKeychainItemImport: Unknown format in import.
> ```

Note the first assumption was wrong because the new Fastlane version just showed a problem that was already happening but not properly addressed (something I've discovered later across this issue). Also, the other CI server is not ephemeral and it has the certs installed in the keychain, so it's not a good example to compare.

**Comment by philipphofmann** [comment]
We still see this happening in the https://github.com/getsentry/sentry-cocoa repo: `Could not install WWDR certificate` Rerunning the GH action solves the problem. We didn't see this error when using Fastlane `2.210.1`.

**Comment by pschipitsch** [comment]
Hitting this with Fastlane `2.212.1`

**Comment by keithkakumanu** [comment]
> I think the root cause of this issue is that fastlane is looking for the certificate in the Login Keychain and the cert is installed in the System Keychain
> 
> Failing `security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/vagrant/Library/Keychains/login.keychain-db`
> 
> Working `security find-certificate -a -c 'Apple Worldwide Developer Relations'`
> 
> When fastlane fails to find the cert it downloads it from Apple which intermittently fails

Same for us using it on `CircleCI`. The cert is stored in the `system` keychain.
<img width="1021" alt="Screen Shot 2023-03-24 at 9 56 57 am" src="https://user-images.githubusercontent.com/93102860/227382748-a2f90db1-eb0e-4400-9636-e6d15fdf5a25.png">

**Comment by bartpowers** [comment]
I was able to workaround this intermittent failure by adding a bash step to our pipeline that runs before the fast lane commands:
```
tmpfile=$(mktemp /tmp/wwdr-cert.cer)
          curl -f -o $tmpfile https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer && security import $tmpfile /Users/runner/Library/Keychains/login.keychain-db
```

Swap out `/Users/runner/Library/Keychains/login.keychain-db` for the path to your pipeline's `login.keychain-db`

**Comment by farshadtx** [comment]
Is there any update on this?! not fixed yet.

**Comment by mluisbrown** [comment]
We're also seeing this issue intermittently, why is the issue closed?

**Comment by jase88** [comment]
@joshdholtz can the ticket be opened again? Or does it have to be a new issue?

Since many are reporting downgrading to `2.210.1` fixes the issue it might be interesting what changed since then.

**Comment by xucian** [comment]
> We're also seeing this issue intermittently, why is the issue closed?

same question

**Comment by rickpasetto** [comment]
I'm still seeing this as well

**Comment by shichen85** [comment]
Just got this error today.

**Comment by gabsn** [comment]
same

**Comment by vjjft** [comment]
Comparing `2.210.1` and `2.211.0` shows a likely relevant change[^1]. Finding[^2] the relevant commit, it seems[^3] that there had been a bug (from 6 years ago[^4] that would ignore the result of the fetch (via `curl`) of the certificates. It seems this bug was possibly/probably benign.

[^1]: https://github.com/fastlane/fastlane/compare/2.210.1...2.211.0#diff-605d869e313da5ef9f0c402845a4911eed5f8afc4ef45382210f237fceae498aR169
[^2]: https://github.com/fastlane/fastlane/blame/2.211.0/fastlane_core/lib/fastlane_core/cert_checker.rb#L169
[^3]: https://github.com/fastlane/fastlane/commit/4b3008914ff809850d7b8bf772e1a3363213ed18#:~:text=concatenation%0A%0A*%20Fix%20some-,WWDR%20certificates,-test%20failure%0A%0AThis
[^4]: 
https://github.com/fastlane/fastlane/blame/7de28ca20ebbee386728ea8eb35838a67375a9ca/fastlane_core/lib/fastlane_core/cert_checker.rb#L169

**Comment by shtefanilie** [comment]
Getting this one too on `2.212.0`.

**Comment by ethernetz** [comment]
I hit it as well using Github actions

**Comment by biodranik** [comment]
The same error with 2.212.2

**Comment by henderson2350** [comment]
> The same error with 2.212.2 while using [GitHub Actions](https://github.com/organicmaps/organicmaps/actions/runs/4849263827/jobs/8641091314).

same here

**Comment by pchelnikov** [comment]
Got this today for the first time. `GitHub Actions`, `macos-13` runner, fastlane `2.211.0`.

**Comment by ivan901229** [comment]
Getting this issue too on 2.211.0.

**Comment by lukeredpath** [comment]
Just got this for the first time today. CircleCI, Match 2.211.0.

**Comment by chrisshaddad** [comment]
@pchelnikov, we're currently still facing this issue on `2.212.2`.

Running on GH actions, `macos-13` runner.

**Comment by biodranik** [comment]
We have solved it by manually importing Apple WWDR certificate during the build in the Fastline config.

**Comment by henderson2350** [comment]
> We have solved it by manually importing Apple WWDR certificate during the build in the Fastline config.

how do you do this?

**Comment by mluisbrown** [comment]
@henderson2350 see this comment above for one possible way to do it: https://github.com/fastlane/fastlane/issues/20960#issuecomment-1482985500

**Comment by biodranik** [comment]
Download https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer

Then add the following lines to the Fastfile: https://github.com/organicmaps/organicmaps/pull/5157/commits/8028798ceff0d91992b318e400559a37a034af51

**Comment by csath** [comment]
This is still happening intermittently for me even on fastlane-2.213.0 version in Bitrise.

**Comment by daniil-shumko** [comment]
I started having this issue on GitHub Actions. The [runner](https://github.com/actions/runner-images/blob/macos-13/20230611.2/images/macos/macos-13-Readme.md#tools) has `fastlane 2.213.0` installed. 
I am using Xcode 14.3.1

**Comment by selobu** [comment]
We are using Fastlane 2.210.1 and XCODE 14.1  but the error randomly occurs

**Comment by Legoless** [comment]
Like other's are saying, still an issue with latest 2.213.0 and Xcode 14.3.1 on GitHub Actions. :/

**Comment by nikolaykasyanov** [comment]
Even with one of the suggested workarounds (installing the G3 certificate manually) + the retry environment variable, we're still getting "Could not install WWDR certificate" occasionally, when Fastlane tries to download the other ones:

```
$ security default-keychain -d user
▸     "/Users/vagrant/Library/Keychains/login.keychain-db"
 Installing WWDR Cert: curl --http1.1 --retry 3 --retry-all-errors -f -o /var/folders/69/qgnyrbkx23dbp0p6p_b6bc9m0000gn/T/AppleWWDRCAG2.cer20230705-26839-qa9p96 https://www.apple.com/certificateauthority/AppleWWDRCAG2.cer && security import /var/folders/69/qgnyrbkx23dbp0p6p_b6bc9m0000gn/T/AppleWWDRCAG2.cer20230705-26839-qa9p96 -k /Users/vagrant/Library/Keychains/login.keychain-db
▸ 1 certificate imported.
▸ 100   763  100   763    0     0  13287      0 --:--:-- --:--:-- --:--:-- 13625
$ security default-keychain -d user
▸     "/Users/vagrant/Library/Keychains/login.keychain-db"
 Installing WWDR Cert: curl --http1.1 --retry 3 --retry-all-errors -f -o /var/folders/69/qgnyrbkx23dbp0p6p_b6bc9m0000gn/T/AppleWWDRCAG4.cer20230705-26839-55p8ad https://www.apple.com/certificateauthority/AppleWWDRCAG4.cer && security import /var/folders/69/qgnyrbkx23dbp0p6p_b6bc9m0000gn/T/AppleWWDRCAG4.cer20230705-26839-55p8ad -k /Users/vagrant/Library/Keychains/login.keychain-db
▸ 100  1113  100  1113    0     0  23626      0 --:--:-- --:--:-- --:--:-- 24733
▸ security: SecKeychainItemImport: Unknown format in import.
 Failed to install WWDR Certificate, checking output to see why
```

**Comment by juan-vg** [comment]
The actual workaround would be to download the 6 certs

- https://developer.apple.com/certificationauthority/AppleWWDRCA.cer
- https://www.apple.com/certificateauthority/AppleWWDRCAG2.cer
- https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer
- https://www.apple.com/certificateauthority/AppleWWDRCAG4.cer
- https://www.apple.com/certificateauthority/AppleWWDRCAG5.cer
- https://www.apple.com/certificateauthority/AppleWWDRCAG6.cer

and install them in `login.keychain`

**Comment by fruitcoder** [comment]
why was this issue closed? Apparently, it's still a problem. Or can anyone confirm it's resolved in the latest fastlane release?

**Comment by Legoless** [comment]
Nope, still an issue for us (2.213.0), so we use the workaround.

**Comment by fruitcoder** [comment]
@Legoless which of the many workarounds do you mean and how do they work on a CI? Should we download and install them on every build via fastlane or download them on our machine and upload them to the CI which installs them on launch (my example is bitrise). I don't know which keychain certificates are installed into by default.

**Comment by biodranik** [comment]
@fruitcoder Get Apple certificates and install them in Fastfile, for example, like that: https://github.com/organicmaps/organicmaps/blob/master/xcode/fastlane/Fastfile#L13

**Comment by fruitcoder** [comment]
Thanks for the input. I tried a manual script that downloads the certs and installs them in the keychain. Will have to find out if it works for the next build. Otherwise, I'll try your solution in fastlane directly

**Comment by Legoless** [comment]
> @Legoless which of the many workarounds do you mean and how do they work on a CI? Should we download and install them on every build via fastlane or download them on our machine and upload them to the CI which installs them on launch (my example is bitrise). I don't know which keychain certificates are installed into by default.

@fruitcoder We install the certificates ourselves before in addition to our distribution certificate that we use to sign the app on CI. Similar procedure to:
```
import_certificate(
        certificate_path: 'fastlane/certificates/AppleWWDRCAG3.cer',
        keychain_name: ENV['MATCH_KEYCHAIN_NAME'],
        keychain_password: ENV['MATCH_KEYCHAIN_PASSWORD']
    )
```

**Comment by zotrix** [comment]
```
❯ security import /tmp/AppleWWDRCAG6.cer20230720-19211-1xp81x  -k test5-db
security: SecKeychainItemImport: Unknown format in import.

 mv /tmp/AppleWWDRCAG6.cer20230720-19211-1xp81x /tmp/AppleWWDRCAG6.cer20230720-19211-1xp81x.cer

security import /tmp/AppleWWDRCAG6.cer20230720-19211-1xp81x.cer  -k test6-db
1 certificate imported.
```

or with format
```
❯ security import /tmp/AppleWWDRCAG6.cer20230720-19211-1xp81x -f pkcs7 -k test5-db
1 certificate imported.
```

I think problem near 
```
      file = Tempfile.new(File.basename(url))
```

need save extension for temfile. 

Something like this
```
base = File.basename(url)
file = Tempfile.new([base, File.extname(base)])
```

**Comment by doyeka** [comment]
encountered the same error in Github Actions using 2.214.0. downgrading to 2.213.0 fixed it for me, but given everyone else in this thread, I expect it will fail intermittently.

I'll probably resort to the [mentioned workaround](https://github.com/fastlane/fastlane/issues/20960#issuecomment-1482985500) if I face any more issues.

@joshdholtz please take another look beyond adding retries.

**Comment by snowtema** [comment]
Same problem, downgrading fastlane was unsuccessful

**Comment by nekrich** [comment]
@joshdholtz , @PaulTaykalo in #21442 figured out a real reason and it looks like a bug in security import.
Executing `security import AppleWWDRCAG2.cerXXXXXXXXXXX-X-p8asd` will produce an error `security: SecKeychainItemImport: Unknown format in import`.

The workaround it to preserve .cer file extension.

**Comment by rabbitinspace** [comment]
I still has this issue when using a non-default keychain, so I opened https://github.com/fastlane/fastlane/pull/21578 which fixes it in our CI

**Comment by williamdes** [comment]
Fastlane: 2.220.0

Warning: it needs the keychain to be named "login.keychain"

Here is the solution that worked for me on my github-runner:
```sh
# Install cert
# Download
# https://developer.apple.com/certificationauthority/AppleWWDRCA.cer
# https://www.apple.com/certificateauthority/AppleWWDRCAG2.cer
# https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer
# https://www.apple.com/certificateauthority/AppleWWDRCAG4.cer
# https://www.apple.com/certificateauthority/AppleWWDRCAG5.cer
# https://www.apple.com/certificateauthority/AppleWWDRCAG6.cer
# See: https://github.com/fastlane/fastlane/issues/20960#issuecomment-1621931850
security create-keychain /Users/runner/Library/Keychains/login.keychain-db
security default-keychain -s /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCA.cer -k /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCAG2.cer -k /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCAG3.cer -k /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCAG4.cer -k /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCAG5.cer -k /Users/runner/Library/Keychains/login.keychain-db
security import ./AppleWWDRCAG6.cer -k /Users/runner/Library/Keychains/login.keychain-db
security find-identity /Users/runner/Library/Keychains/login.keychain-db
# No identities are found (https://github.com/fastlane/fastlane/issues/20960)
# Request a certificate (https://developer.apple.com/account/resources/certificates/list)
openssl req -subj "/emailAddress=xx@xxx.xx/CN=User Name/C=XX" -out code_signing.csr -new -newkey rsa:2048 -nodes -keyout code_signing.key
# Display it
openssl req -in code_signing.csr -text -noout
# Upload the code_signing.csr to the Apple website and get the certificate, rename it to code_signing.cer
# Delete it since it is not needed
rm code_signing.csr
security import /Users/runner/code_signing.key /Users/runner/Library/Keychains/login.keychain-db
security import /Users/runner/code_signing.cer /Users/runner/Library/Keychains/login.keychain-db
# Check it worked
security find-identity  /Users/runner/Library/Keychains/login.keychain-db
```
